lib/nerves_hub/console_channel.ex

```diff
@@ -174,12 +174,25 @@ defmodule NervesHub.ConsoleChannel do
     state
   end
 
-  # All other requests push IO Request over the socket. 
+  # All other requests push IO Request over the socket.
   defp io_request(from, reply_as, {:get_line, encoding, data} = req, state) do
     Channel.push_async(state.channel, "get_line", %{encoding: encoding, data: data})
     %{state | request: {from, reply_as, req}}
   end
 
+  defp io_request(from, reply_as, {:put_chars, encoding, module, function, args}, state) do
+    # Judicious try block so we can return back the error tuple
+    # as defined in http://erlang.org/doc/apps/stdlib/io_protocol.html#output-requests
+    try do
+      data = apply(module, function, args)
+      io_request(from, reply_as, {:put_chars, encoding, data}, state)
+    rescue
```

Comment:
i think doing `catch` here might be stricter in getting every error.

Reply:
I read the note in the Erlang guide, and it sounded like they wanted you to catch thrown values. I.e., to use `catch`, but don't handle exits or errors.

Reply:
Also, fwiw, Elixir's `StringIO` doesn't catch or rescue. However, the files I looked at in Erlang did catch thrown values.

Reply:
I'm going off of this:
> If the function returns anything else than a binary or list, or throws an exception, an error is to be sent back to the client.
>The I/O server replies to the client with an io_reply tuple, where element Reply is one of:
>ok
>{error, Error}

In this case, we are the "Server" expected to apply the function and return back `ok`, or `{error, Error}`. Which, to me, seems like it wants any exception sent back to the client.

We could just not handle it and assume the exception would crash the process and restart the console or try to handle it like expected. I think it's 6's and opted for trying to manage according to Erlang doc. Especially because if we _don't_ return one of those 2 values, it totally messes up the IEx process.

I think you're right about `catch` now that I'm looking at it again.
> The catch clause can be used to catch thrown values, exits, and errors.

Reply:
Yes, please change so that we're consistent with the Erlang docs and Erlang implementations. I think that the Elixir one is an oversight or an observation that anything throwing on strings is deserves the pain.

Reply:
Updated to use `catch`